### DISCUSSION TITLE
Privacy-Preserving Holder Binding

### DISCUSSION THREAD

**Discussion by ad-Orange** [e-id-admin/open-source-community#30] [General]
Maybe there is another option: [BBS#](https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/discussions/211#discussioncomment-9882200)

**Comment by bjafs** [comment]
Thanks for the link. As you might have seen in our [**Technical Roadmap** ](https://github.com/e-id-admin/open-source-community/blob/main/tech-roadmap/tech-roadmap.md) we have a specific section on privacy preserving holder-binding. 
We will gladly take a look at the link you shared with us and the corresponding paper. Is there any open-source implementation yet which you could refer us to?

Best Regards,

**Reply by ad-Orange** [reply]
Yes, we have seen this section about "privacy-preserving holder binding" and this is exactly the reason why we posted this.
We have implemented the protocol on Android Strongbox for now but unfortunately don't have an open-source implementation yet. However, given the amount of requests we get, we are working on a reference implementation.
In the meantime, you can have a look at what dock.io [have done](https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/issues/193#issuecomment-2206354831) (it's basically a [partial implementation](https://github.com/docknetwork/crypto/tree/main/kvac/src/bbdt_2016) of BBS# based on a paper we co-authored in 2016). We also noticed that Christian Khalo seems to have [implemented it](https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/discussions/211#discussioncomment-9898689) successfully.
Finally, another [link](https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/discussions/211#discussioncomment-9896573) that might clarify a few things in terms of privacy-preserving holder binding.
I hope you will find all that interesting.

**Reply by ad-Orange** [reply]
Hello,
The dock.io partial implementation we were referring to before is quickly growing into a full blown implementation. Please see [here](https://github.com/eu-digital-identity-wallet/eudi-doc-architecture-and-reference-framework/discussions/211#discussioncomment-10173580) and don’t hesitate to ask for more details if needed.

**Reply by bit-jniestroj** [reply]
Thank you for the heads up about the dock.io implementation. We will have a deeper look into it.

Best Regards

**Comment by ad-Orange** [comment]
Hello,
Did you have the opportunity to look into it ?

**Reply by bjafs** [reply]
Hey there,
Due to absences, we can’t provide a detailed feedback yet. We will get back to you in due course.
Best Regards
